### ISSUE TITLE
Crash with invalid test filter regex expression

### ISSUE THREAD

**Issue by victorusu** [eth-cscs/reframe#909]
When I attempt to run the following regex expression in order to filter test
```
reframe -n *romacs -l
```

I get the stack trace
```
...
.../bin/reframe: unexpected error: nothing to repeat at position 0
Traceback (most recent call last):
  File ".../frontend/cli.py", line 425, in main
    checks_matched = filter(filters.have_name('|'.join(options.names)),
  File ".../frontend/check_filters.py", line 8, in have_name
    regex = re.compile(patt)
  File ".../3.6.6/lib/python3.6/re.py", line 233, in compile
    return _compile(pattern, flags)
  File ".../3.6.6/lib/python3.6/re.py", line 301, in _compile
    p = sre_compile.compile(pattern, flags)
  File ".../3.6.6/lib/python3.6/sre_compile.py", line 562, in compile
    p = sre_parse.parse(p, flags)
  File ".../3.6.6/lib/python3.6/sre_parse.py", line 855, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)
  File ".../3.6.6/lib/python3.6/sre_parse.py", line 416, in _parse_sub
    not nested and not items))
  File ".../3.6.6/lib/python3.6/sre_parse.py", line 616, in _parse
    source.tell() - here + len(this))
sre_constants.error: nothing to repeat at position 0
```

I do not know what would be the best behaviour. Support this UNIX like filter or just print that this is not a valid Python regex.

**Comment by vkarak** [comment]
@victorusu We state explicitly that we support Python regex syntax only. No plans to mix both syntaxes.

**Comment by teojgo** [comment]
@victorusu which version of ReFrame are you using? Because with the latest one I get:
```
invalid regex: '*romacs'
```

**Comment by vkarak** [comment]
@teojgo So I remember correctly that I have somehow dealt with this issue! Long vacations made me forget completely that I have fixed that! It was fixed in 2.18 (#757).

**Comment by vkarak** [comment]
Closing, duplicate of #730. Fixed in #757.
